engines/group/app/controllers/group/api/v1/groups_controller.rb

```diff
@@ -5,31 +5,42 @@ class GroupsController < ApplicationController
 
         before_action :authenticate
         before_action :find_and_authorize_group, only: [:show, :update, :destroy]
+        before_action :initialize_side_effects, only: [:create]
+        after_action :add_side_effects_links, only: [:create]
 
+        # GET api/v1/groups
+        #
         def index
           render json: GroupsSerializer.new(current_user.groups)
         end
 
+        # GET api/v1/groups/:id
+        #
         def show
           render json: GroupSerializer.new(@group)
         end
 
+        # POST api/v1/groups
+        #
         def create
           group = ::Group::Group.new(group_params)
-          group_creation = GroupCreation.new(group, current_user)
-          if group_creation.create
+          group_creator = GroupCreator.new(group, current_user)
+          if group_creator.create
+            @side_effects << group_creator.side_effects
```

Comment:
I don't like side effects :) It sounds like a bad thing. This is a resource link. Why we just call `resource_links`?